apps/aecore/src/aec_chain_state.erl

```diff
@@ -38,6 +39,7 @@
                     , 'max_snapshot_height' => pos_integer()
                     , 'sparse_snapshots_interval' => pos_integer()
                     , 'keep_all_snapshots_height' => pos_integer()
+                    , 'genesis_block_hash' => 'undefined' | binary()
```

Comment:
Can this ever be undefined?

Reply:
I see this is undefined because of how a new state is created:
* When [state is created knowing genesis block](https://github.com/aeternity/epoch/pull/399/files#diff-675152b2278724570b642ab179e889e3R68), genesis block could be passed to function initializing this state that would compute this field - hence no need of undefined for this case;
* When [state is created from persisted chain](https://github.com/aeternity/epoch/pull/399/files#diff-675152b2278724570b642ab179e889e3R73), it looks like that the persisted state that is argument of init (`Chain, TopBlockHash, TopBlockStateTrees}`) shall contain also `GenesisBlockHash`.

I need to understand whether when starting from persisted state:
* This field ever get set;
* This causes start time to be longer.

Reply:
Could you please hardcode expectation at the end of init that this field is a binary (hence undefined). This shall be done in two points:
* [Init from genesis block](https://github.com/aeternity/epoch/pull/399/files#diff-675152b2278724570b642ab179e889e3R69) - this shall be straightforward;
* [Init from persisted chain](https://github.com/aeternity/epoch/pull/399/files#diff-675152b2278724570b642ab179e889e3R74) - this would hardcode expectation that persisted chain must contain genesis header (that seems reasonable).

Optionally, in init that start from persisted chain pass genesis block (https://github.com/aeternity/epoch/pull/399/files#diff-b98da3b40a4c433bc375929f51069cb3R87) and check that the genesis header hash from the persisted chain is the same as the one from the expected genesis.

Reply:
> Optionally, in init that start from persisted chain pass genesis block (https://github.com/aeternity/epoch/pull/399/files#diff-b98da3b40a4c433bc375929f51069cb3R87) and check that the genesis header hash from the persisted chain is the same as the one from the expected genesis.

Opened https://www.pivotaltracker.com/story/show/153184647 for this optional item.

Reply:
> Can this ever be undefined?

Opened https://www.pivotaltracker.com/story/show/153184755 for tracking this.